### PR TITLE
Query account keys and check signatures.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Closes https://github.com/Concordium/concordium-base/issues/388

Depends on
- [x] https://github.com/Concordium/concordium-base/pull/404/

## Changes

Add two new operations that can be invoked via `invoke`, retrieving account keys and checking account signatures.

- [x] Analyze performance

Benchmarking running a node locally suggests the cost assignments are good. The worst case seems to be around 0.7s for block execution with 3_000_000 NRG.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by abizjak** [comment]
> @limemloh @Bargsteen In the interest of saving time I am asking for review before having done proper testing and benchmarking myself. So I can work on that in parallel.

I have added a unit test now for the new functionality. I did a bit of testing for performance, but I need to verify a few more cases, but I believe it is adequate overall.

File: concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/AccountSignatureChecks.hs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    (InvokeHelpers.PersistentModuleInterfaceV GSWasm.V1, WasmModuleV GSWasm.V1) ->
+    Helpers.PersistentBSM pv (Types.ContractAddress, PersistentBlockState pv)
+initContract =
+    InvokeHelpers.initContractV1
+        (Helpers.accountAddressFromSeed seed)
+        (InitName "init_contract")
+        emptyParameter
+        0
+
+-- |Invoke the entrypoint @@ on contract @a@.
```
Comment (Bargsteen) [inline concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/AccountSignatureChecks.hs:74]:
```suggestion
-- |Invoke the entrypoint @get_keys@ on contract @a@.
```

**Review by Bargsteen — APPROVED** [review]
Looks good!

**Review by limemloh — APPROVED** [review]

File: concordium-consensus/src/Concordium/Scheduler/WasmIntegration/V1.hs [hunk 2]
```diff
@@ -399,6 +407,15 @@ data InvokeMethod
         }
     | -- |Query the CCD/EUR and EUR/NRG exchange rates.
       QueryExchangeRates
+    | -- |Check the signature using account keys
+      CheckAccountSignature
+        { imcasAddress :: !AccountAddress,
+          imcasPayload :: !BS.ByteString
```
[the remark this answers is not in the document]
Comment (abizjak) [reply concordium-consensus/src/Concordium/Scheduler/WasmIntegration/V1.hs:413]: Added

**Pull request merged**